lib/srv/monitor.go

```diff
@@ -111,17 +113,20 @@ func NewMonitor(cfg MonitorConfig) (*Monitor, error) {
 	if err := cfg.CheckAndSetDefaults(); err != nil {
 		return nil, trace.Wrap(err)
 	}
-	return &Monitor{
+	m := &Monitor{
 		MonitorConfig: cfg,
-	}, nil
+	}
+
+	return m, nil
 }
 
-// Monitor monitors connection activity
-// and disconnects connections with expired certificates
-// or with periods of inactivity
+// Monitor monitors the activity on a single connection and disconnects
+// that connection if the certificate expires or after
+// periods of inactivity
 type Monitor struct {
 	// MonitorConfig is a connection monitor configuration
 	MonitorConfig
+	Shell ssh.Channel
```

Comment:
We need to keep in mind other protocols too I think. E.g. databases.

Can we make this interface a bit more generic? Rather than embedding ssh channel directly, maybe accept some kind of writer interface that will be responsible for writing the idle message to the session? Database access uses the same monitor and it will be able to initialize it with, say, a writer that will send a message to a database client, where it's supported.